### ISSUE TITLE
Question on setting nodeAffinity using a set array in helm_release

### ISSUE THREAD

**Issue by cychoi34** [hashicorp/terraform-provider-helm#1433]
## Terraform version, Kubernetes provider version and Kubernetes version

```
Terraform version: Terraform v1.9.2
Helm Provider version: v3.14.3+gf03cc04
Kubernetes version: 1.29
```
## Terraform configuration

```hcl
resource "helm_release" "nginx-ingress-external" {

  name      = "blahblah"
  chart     = "argo-cd"
  version   = "0.35.4"
  namespace = "${var.argocd}"

...
  set {
    name  = "controller.replicaCount"
    value = "2"
  }

  set {
    name  = "controller.minAvailable"
    value = "1"
  }
...

  set {
    name  = "controller.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution[0].labelSelector.matchExpressions[0].key"
    value = "Name"
  }

  set {
    name  = "controller.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution[0].labelSelector.matchExpressions[0].operator"
    value = "In"
  }

  set {
    name  = "controller.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution[0].labelSelector.matchExpressions[0].values[0]"
    value = "something"
  }
}
```

## Question

Hi, I want the ArgoCD pods to run on the node group.
Other pods are all good for nodeAffinity setup.
In the second image, you may see that the row (tfstate file) indicates that nodeAffinity is adjusted, but it's not actually applied.
From the first image, you can see there's only podAntiAffinity.

![image01](https://github.com/user-attachments/assets/b03e1b5b-e5e3-4c0f-b401-29e6359e7511)
![image02](https://github.com/user-attachments/assets/70a75086-ee45-4179-b2ec-067fe151b38f)

**Comment by BBBmau** [comment]
@cychoi34 it seems like this could be due to some syntax typos. Can you attempt setting the values through the helm CLI to compare the results?

**Comment by cychoi34** [comment]
@BBBmau Yeah, that’s right. It seems a bit clearer now. I was using `affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key=Name` instead of `controller.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key=Name`, 
I could found it using `helm template` command.
so you’re right. It was a syntax typo.
Thanks for your time, BBBmau!
![image](https://github.com/user-attachments/assets/52d3e519-0d22-4db6-aa03-3c44e071b44f)

**Comment by cychoi34** [comment]
Hi, @BBBmau I have a further question here. 
It seems it's been adjusted at the StatefulSet but not at the Pod level (even for the Deployment)
How can I solve this?

**Comment by BBBmau** [comment]
@cychoi34 you may want to refer to the Argo chart, from my initial investigation I didn't come across anything helpful as to why it wouldn't adjust to the pod level.
